server/src/main/java/org/opensearch/gateway/remote/model/RemotePinnedTimestamps.java

```diff
@@ -0,0 +1,140 @@
+import org.opensearch.common.remote.RemoteWriteableBlobEntity;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.compress.Compressor;
+import org.opensearch.index.remote.RemoteStoreUtils;
+import org.opensearch.repositories.blobstore.ChecksumWritableBlobStoreFormat;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.opensearch.gateway.remote.RemoteClusterStateUtils.DELIMITER;
+
+/**
+ * Wrapper class for uploading/downloading {@link RemotePinnedTimestamps} to/from remote blob store
+ *
+ * @opensearch.internal
+ */
+public class RemotePinnedTimestamps extends RemoteWriteableBlobEntity<RemotePinnedTimestamps.PinnedTimestamps> {
+    private static final Logger logger = LogManager.getLogger(RemotePinnedTimestamps.class);
+
+    /**
+     * Represents a collection of pinned timestamps and their associated pinning entities.
+     * This class is thread-safe and implements the Writeable interface for serialization.
+     */
+    public static class PinnedTimestamps implements Writeable {
+        private final Map<Long, List<String>> pinnedTimestampPinningEntityMap;
+
+        public PinnedTimestamps(Map<Long, List<String>> pinnedTimestampPinningEntityMap) {
+            this.pinnedTimestampPinningEntityMap = new ConcurrentHashMap<>(pinnedTimestampPinningEntityMap);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeMap(pinnedTimestampPinningEntityMap, StreamOutput::writeLong, StreamOutput::writeStringCollection);
+        }
+
+        public static PinnedTimestamps readFrom(StreamInput in) throws IOException {
+            return new PinnedTimestamps(in.readMap(StreamInput::readLong, StreamInput::readStringList));
+        }
+
+        /**
+         * Pins a timestamp against a pinning entity.
+         *
+         * @param timestamp The timestamp to pin.
+         * @param pinningEntity The entity pinning the timestamp.
+         */
+        public void pin(Long timestamp, String pinningEntity) {
+            logger.debug("Pinning timestamp = {} against entity = {}", timestamp, pinningEntity);
+            pinnedTimestampPinningEntityMap.computeIfAbsent(timestamp, k -> new ArrayList<>()).add(pinningEntity);
+        }
+
+        /**
+         * Unpins a timestamp for a specific pinning entity.
+         *
+         * @param timestamp The timestamp to unpin.
+         * @param pinningEntity The entity unpinning the timestamp.
+         */
+        public void unpin(Long timestamp, String pinningEntity) {
+            logger.debug("Unpinning timestamp = {} against entity = {}", timestamp, pinningEntity);
```

Comment:
if someone tries to unpin a timestamp and its not pinned(not present in the pinnedTimestampPinningEntityMap) do you think we should log a warning ?